### PR TITLE
Fixed cpu frequency retrieval and clock cycle for IBM Power systems

### PR DESCRIPTION
**Pull request by npanpaliya**

### CODE REVIEW

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you very much for the PR and I'm so sorry for my delay!

**Comment by npanpaliya** [comment]
Thank you @penpornk for the comments and sorry for the delay as I was occupied in other tasks. I've addressed all your suggestions.

File: tensorflow/core/platform/profile_utils/cpu_utils.h [hunk 1]
```diff
@@ -93,6 +93,27 @@ class CpuUtils {
...
     return DUMMY_CYCLE_CLOCK;
+#elif defined(__powerpc64__) || defined(__ppc64__)
+    uint64 __t;
+    __asm__ __volatile__ (	
+      "mfspr %0,268"
+      : "=r" (__t));
+    return __t;
+
+#elif defined(__powerpc__) || defined(__ppc__) 
+    uint64 result=0;
```
Comment (penpornk) [inline tensorflow/core/platform/profile_utils/cpu_utils.h]: Nit: `result` is not used. We can remove it. Same for line 115.

**Review by penpornk — CHANGES_REQUESTED** [review]
@npanpaliya No worries at all and thank you for the fixes! Would you mind making one more minor change?

**Review by penpornk — APPROVED** [review]
Thank you very much again for contributing! :)

**Comment by npanpaliya** [comment]
@penpornk Thank you for reviewing and merging this.

**Pull request merged**